app/src/main/java/com/example/weatherapp/data/Service.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.weatherapp.data
+
+import com.example.weatherapp.model.WeatherData
+import retrofit2.Call
+import retrofit2.http.GET
+
+interface Service {
+    @GET(GET_CITY)
+    suspend fun weatherCityByName(): Call<WeatherData>
```

Comment:
Refactor:

Make use of Retrofit.Response object wrapper. It will provide the convenient methods
isSuccessfull
along with the possibility to check the request code.

Reply:
got it, will fix it